processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -215,13 +214,49 @@ public GroupByQueryQueryToolChest(
       } else {
         finalizingResults = subqueryResult;
       }
-
-      return groupByStrategy.processSubqueryResult(subquery, query, resource, finalizingResults);
+      return groupByStrategy.processSubqueryResult(newSubquery, query, resource, finalizingResults);
     } else {
       return groupByStrategy.mergeResults(runner, query, context);
     }
   }
 
+  public static boolean shouldPushDownQuery(GroupByQuery q)
+  {
+    return QueryContexts.parseBoolean(q, GroupByQueryConfig.CTX_KEY_FORCE_PUSH_DOWN_NESTED_QUERY, false);
```

Comment:
Can be simplified to `return q.getContextBoolean(GroupByQueryConfig.CTX_KEY_FORCE_PUSH_DOWN_NESTED_QUERY, false);` and then https://github.com/druid-io/druid/pull/5471/files#diff-76d2b703319f06a8103ebce8a58ef7fcR221 is not necessary.